pkg/smokescreen/smokescreen.go

```diff
@@ -475,6 +475,15 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 
 		sctx.logger.WithField("url", req.RequestURI).Debug("received HTTP proxy request")
 
+		// Call the custom request handler if it exists
+		if config.CustomRequestHandler != nil {
+			err = config.CustomRequestHandler(req)
+			if err != nil {
+				pctx.Error = denyError{err}
```

Comment:
WDYT about creating a new error type here?

Reply:
I don't feel too strongly about this, but I was pretty intuitively happy with denyErrors containing the error returned by the custom handler. I'm ok with keeping as is, unless you have stronger opinions.